### DISCUSSION TITLE
PHP/Laravel

### DISCUSSION THREAD

**Discussion by svivier-gp-aviva** [superstreamlabs/memphis#1656] [Ideas]
Hello 😊
Do you plan to develop a client for PHP ? and specifically for Laravel ? to consume messages.
Thanks !

**Comment by g41797** [comment]
DIY
see [PHP gRPC client](https://github.com/g41797/memphisgrpc/tree/main/_generated/php/Pb/Memphisphp)
it contains generated code for memphis grpc adapter
[Supported APIs:](https://github.com/g41797/memphisgrpc/tree/main/proto)
```proto
service AdapterService {
	rpc CreateStation(CreateStationRequest) returns (Status) {}
	rpc DestroyStation(DestroyStationRequest) returns (Status) {}
	rpc Produce(stream ProduceMessages) returns (Status) {}
	rpc Consume(stream ConsumeMessages) returns (stream ConsumeResponse) {}
}
```
you can write thin php wrapper for consumer
if you need help on adapter side, let me know
